### PR TITLE
Version Updates

### PR DESCRIPTION
**Pull request by masonds**
## what
* Asdf updated from 0.8.1 to 0.10.2
* AWS CLI updated from 1.18.108 to 2.8.4
* ContainerD updated from 1.2.13 to 1.6.8
* Docker updated from 19.03.13 to 20.10.9
* Docker Compose updated from 1.26.2 to 2.12.0
* FOSSA CLI updated from 1.0.30 to 3.4.11
* Go updated from 1.16.7 to 1.19.2
* Go Task updated from 3.0.0 to 3.17.0
* Go Releaser updated from 0.151.1 to 1.12.3
* Go-critic updated from 0.6.3 to 0.6.5
* Golangci-lint updated from 1.41.1 to 1.50.0
* Gomplate updated from 3.7.0 to 3.11.0
* Hadolint updated from 1.18.0 to 2.10.0
* Helm updated from 3.6.3 to 3.10.1
* Helm Chart Releaser updated from 1.1.1 to 1.4.1
* Helm-diff updated from 3.1.3 to 3.6.0
* Helm-git updated from 0.11.1 to 0.13.0
* Helm-s3 updated from 0.10.0 to 0.14.0
* Helmfile updated from 0.138.2 to 0.147.0
* Java updated from 11.0.8+10 to 19.0.0+36
* Kubectl updated from 1.18 to 1.25.3
* Maven updated from 3.6.3 to 3.8.6
* NodeJS updated from 12.18.3 to 18.11.0
* Pipenv updated from 2020.11.15 to 2022.10.12
* Podman updated from 3.3.1 to 4.2.0
* Pre-commit updated from 2.7.1 to 2.20.0
* Python updated from 3.8.2 to 3.10.8
* Serverless updated from 3.3.0 to 3.23.0
* Shellcheck updated from 0.7.2 to 0.8.0
* Shellspec updated from 0.22.0 to 0.28.1
* Skopeo updated from 1.4.2 to 1.9.2
* Sonar Scanner updated from 4.6.2.2472 to 4.7.0.2747
* Terraform updated from 0.15.4 to 1.3.3
* Terraform-docs updated from 0.12.1 to 0.16.0
* TFlint updated from 0.28.1 to 0.41.0
* TFspec updated from 0.39.37 to 1.28.0
* Updated spec files as needed

## why
* Generally bringing included tools up to current versions to benefit from improved functionality, security, and stability.

### CODE REVIEW

File: src/docker/Dockerfile [hunk 1]
```diff
@@ -328,47 +330,48 @@ ENV GOPATH="/home/anvil/go"
 ENV PATH="${PATH}:${GOPATH}/bin"
 
 # Install golangci-lint. Get versions using asdf list all golangci-lint
-ARG GOLANGCI_LINT_VERSION="1.41.1"
+ARG GOLANGCI_LINT_VERSION="1.50.0"
 ENV GOLANGCI_LINT_VERSION=${GOLANGCI_LINT_VERSION}
+#   asdf plugin add golangci-lint https://github.com/hypnoglow/asdf-golangci-lint.git
```
Comment (shawju) [inline src/docker/Dockerfile]: Barnacle comment

**Comment by bstascavage** [comment]
LGTM

File: spec/shellspec_spec.sh [hunk 2]
```diff
@@ -1,13 +0,0 @@
-#!/usr/bin/env shellspec
-
-Describe "printVersion()"
```
Comment (shawju) [inline spec/shellspec_spec.sh:3]: What's a good way for us to document/remember that this test was removed? In the future, shellspec upgrades may go unnoticed

**Review by shawju — DISMISSED** [review]
LGTM

**Review by shawju — APPROVED** [review]
LGTM

**Pull request merged**